### DISCUSSION TITLE
How to run command in zsh as soon as wezterm opens?

### DISCUSSION THREAD

**Discussion by crhuber** [wezterm/wezterm#4873] [Q&A] [answered]
### What Operating System(s) are you running on?

macOS

### WezTerm version

20230712-072601

### Ask your question!

How can I run a command in zsh as soon as wezterm opens on mac via the GUI?

I want to keep my shell as zsh so that my .zshrc still runs , and then when zsh runs I want to run a command

I saw in the docs how to use `default_prog` and a launch menu to do it but those arent exactly what I want. I dont want to replace the current shell, and I dont want to have to type a shortcut for my command to run.  For example *in zsh* I just want to run `ls` without launching wezterm from the command line.

Actually the equivalent I want to acheive is something like this :
<img width="466" alt="image" src="https://github.com/wez/wezterm/assets/5857391/9ec335e0-6e5a-4777-a052-6ed3b060d77f">

I want to run the command `fish` once zsh is run so that I can switch between zsh and shell and still my .zshrc environment vars loaded

**Comment by wez** [comment, marked as answer]
```lua
config.default_prog = { 'zsh', '-c', 'command you want goes here' }
```

**Reply by crhuber** [reply]
Thanks!

**Reply by crhuber** [reply]
@wez when I run it this way it seems my .zshrc was not read though

**Reply by gegoune** [reply]
Try adding `-l` to zsh parameters. 

`config.default_prog = { 'zsh', '-l', '-c', 'command you want goes here' }`

**Reply by MarcPhilippeJoly** [reply]
With my configuration, defining `config.default_prog` causes any newly opened tabs to close.
